core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -382,11 +392,37 @@ private[spark] class AppStatusListener(
       }
 
       job.status = event.jobResult match {
-        case JobSucceeded => JobExecutionStatus.SUCCEEDED
-        case JobFailed(_) => JobExecutionStatus.FAILED
+        case JobSucceeded =>
+          appStatusSource.foreach{_.SUCCEEDED_JOBS.inc(1)}
+          JobExecutionStatus.SUCCEEDED
+        case JobFailed(_) =>
+          appStatusSource.foreach{_.FAILED_JOBS.inc(1)}
+          JobExecutionStatus.FAILED
       }
 
       job.completionTime = if (event.time > 0) Some(new Date(event.time)) else None
+
+      for {
+        submissionTime <- job.submissionTime
+        completionTime <- job.completionTime
+      } yield {
+        val localSubmissionTime =
+          LocalDateTime.ofInstant(submissionTime.toInstant, ZoneId.systemDefault)
+        val localCompletionTime =
+          LocalDateTime.ofInstant(completionTime.toInstant, ZoneId.systemDefault)
+        val duration = Duration.between(localCompletionTime, localSubmissionTime)
+        appStatusSource.foreach{_.JOB_DURATION.update(duration.toMillis)}
```

Comment:
Ideally this should be a gauge due to this: https://prometheus.io/docs/instrumenting/writing_exporters/#drop-less-useful-statistics

Reply:
Actually I will transform it to gauge makes more sense.